lib/active_merchant/billing/gateways/go_cardless.rb

```diff
@@ -0,0 +1,183 @@
+require 'json'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class GoCardlessGateway < ActiveMerchant::Billing::Gateway
+      API_VERSION = '2015-07-06'.freeze
+
+      self.test_url = 'https://api-sandbox.gocardless.com'
+      self.live_url = 'https://api.gocardless.com'
+      self.default_currency = 'EUR'
+
+      def initialize(options = {})
+        requires!(options, :access_token)
+        super
+      end
+
+      def purchase(money, token, options = {})
+        post = {
+          payments: {
+            amount: money,
+            currency: options[:currency] || currency(money),
+            description: options[:description],
+            links: {
+              mandate: token
+            }
+          }
+        }
+
+        commit(:post, '/payments', post, options)
+      end
+
+      def store(customer_attributes, bank_account, options = {})
+        res = nil
+        MultiResponse.run do |r|
+          r.process { res = commit(:post, '/customers', customer_params(customer_attributes, options)) }
+          if res.success?
+            r.process { res = create_bank_account(res.params['customers']['id'], bank_account, options) }
+          end
+          if res.success?
+            r.process { create_mandate(res.params['customer_bank_accounts']['id']) }
+          end
+        end
+      end
+
+      def update(customer_attributes, bank_account, options)
+        res = nil
+        MultiResponse.run do |r|
+          r.process { res = commit(:put, "/customers/#{customer_attributes['vault_token']}", customer_params(customer_attributes, options)) }
+          if res.success?
+            r.process { res = create_bank_account(res.params['customers']['id'], bank_account, options) }
+          end
+          if res.success?
+            r.process { create_mandate(res.params['customer_bank_accounts']['id']) }
+          end
+        end
+      end
+
+      def supports_scrubbing?
+        false
```

Comment:
wait does this mean that sensitive data will appear in the data streams?

Reply:
I thought I had read somewhere that going forward scrubbing is a requirement for the AM project.  Could be mistaken

Reply:
On the other hand... AM isnt taking any new gateways.   We have to carry this patch ourselves in any case.   It would be a good thing to do (scrubbing) but maybe not required here.

Reply:
If i'm not mistaken this is not needed, because we don't get any sensitive data from GoCardless. We operate on mandate id and payment id and GoCardless doesn't returns full data (for example after creating a bank account it returns bank id and only last two digits of bank account). Besided that not every gateway has scrubbing enabled (for example: Moneris, Orbital).

Reply:
yeah I was mostly thinking about upstream expectations going forward.   But since is never going upstream, because AM wont take any new ones, we can skip it.